### PR TITLE
utils: Resolve symlinks in copy_icon()

### PR DESCRIPTION
**Pull request by barthalion**
When app-info contains a symlink ${ID}.png pointing to actual
icon file, copy the target instead of storing invalid symbolic
link.

### CODE REVIEW

**Comment by barthalion** [comment]
Related change in appstream-glib: https://github.com/hughsie/appstream-glib/pull/365

File: common/flatpak-utils.c [hunk 1]
```diff
@@ -4210,23 +4210,38 @@ static gboolean
...
   const char *checksum;
+  const char *target_icon_name = icon_name;
 
   if (!ostree_repo_file_ensure_resolved (OSTREE_REPO_FILE(icon_file), NULL))
     {
       g_debug ("No icon at size %s for %s", size, id);
       return TRUE;
     }
 
+  f = g_file_resolve_relative_path (root, g_file_get_path(icon_file) );
```
Comment (alexlarsson) [inline common/flatpak-utils.c:4232]: Why do you need to to this resolve stuff? You should be able to query-info icon_file directly, no?

File: common/flatpak-utils.c [hunk 2]
```diff
@@ -4210,23 +4210,38 @@ static gboolean
...
+  file_info = g_file_query_info (f, OSTREE_GIO_FAST_QUERYINFO,
+                                 G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
+                                 NULL, error);
+
+  if (g_file_info_get_file_type (file_info) == G_FILE_TYPE_SYMBOLIC_LINK)
+  {
+      target_icon_name = g_file_info_get_attribute_byte_string (file_info, "standard::symlink-target");
+      icon_file = g_file_get_child (size_dir, target_icon_name);
```
Comment (alexlarsson) [inline common/flatpak-utils.c:4240]: This however, need to be g_file_resolve_relative_path() to handle symlinks that are not just a filename.

**Comment by owtaylor** [comment]
Looked through the last three commits, and they look OK. I might have had a slight preference for using a annotation rather than a label, but since we've dropped the annotation handling code, I guess that would be more work. And in the end, injecting a magic pseudo-label into the index is not worse than injecting a magic pseudo-annotation - neither has anything to do with the actual contents of the image, and if it's easy for the index generator to do one, it's easy for it to do the other.

**Comment by barthalion** [comment]
Closing for now as it does not make sense until https://github.com/hughsie/appstream-glib/pull/365 lands (or not).

**Pull request closed without merging**